### PR TITLE
feat(config): salt has a module to serialize to TOML

### PR DESCRIPTION
**Pull request by baby-gnu**
The use of a custom module is not required anymore since salt 2017 support ended 2019-12-31.

* telegraf/files/default/telegraf.tmpl.jinja: replace custom module call with `salt['slsutil.serialize']`.

BREAKING CHANGE: `salt['slsutil.serialize']` requires salt >= 2018.3

----

#

### CODE REVIEW

**Comment by baby-gnu** [comment]
It looks like on Centos, the salt-minion does not find the newly installed `toml` python3 library.

For openSUSE, this is an infrastructure problem.

**Comment by baby-gnu** [comment]
> It looks like on Centos, the salt-minion does not find the newly installed `toml` python3 library.

I found the issue in `map.jinja`, the `python3-toml` package was installed for the builtin module but Centos 8 has a `python3-pytoml`.

File: telegraf/map.jinja [hunk 1]
```diff
@@ -58,7 +58,7 @@
 {%-   endif %}
 {%- elif grains.pythonversion[0] == 3 %}
 {%-   if osfinger in ['CentOS Linux-8'] %}
-{%-     do telegraf.update({'toml_pkg': 'python3-toml'}) %}
+{%-     do telegraf.update({'toml_pkg': 'python3-pytoml'}) %}
 {%-   elif osfinger in ['CentOS Linux-7'] %}
```
Comment (myii) [inline telegraf/map.jinja]: @baby-gnu We can just use the `else`:

```suggestion
{%-   if osfinger in ['CentOS Linux-7'] %}
```

**Comment by myii** [comment]
Bit short on time but a few more comments:

* https://github.com/saltstack/salt/commit/e6e442cfc61e7643d233917c303cfc1a93395970
  - `toml` serializer was introduced in `2018.3`
* https://github.com/saltstack/salt/commit/3898ab27c81d5ba7523bd75e23179e621d837102
  - But was only added to the docs for `3000`
* https://travis-ci.org/myii/telegraf-formula/builds/659478502
  - Some instances that may be worth considering

---

Thanks for the review, @n-rodriguez.

**Comment by myii** [comment]
> * `toml` serializer was introduced in `2018.3`

@baby-gnu It would be worth adding this as a `BREAKING CHANGE` message as well.  Two reasons why the minimum version supported is `2018.3`.

**Comment by baby-gnu** [comment]
> * https://travis-ci.org/myii/telegraf-formula/builds/659478502
>     * Some instances that may be worth considering

I think they should be managed in a separate pull request with a rework of the `map.jinja` and the handling of the `python` library.

File: telegraf/map.jinja (hunk 1 above)
Comment (baby-gnu) [reply telegraf/map.jinja]: Thanks. This could be improved later based on `python.jinja` from `libvirt-template`

**Comment by baby-gnu** [comment]
We should not that maybe someday the `toml` library used will change : https://github.com/saltstack/salt/pull/56323

**Comment by myii** [comment]
@baby-gnu Would you mind including this commit to finalise this PR?

* https://github.com/eole/telegraf-formula/commit/db7098ec195033b6e36e0826f3eadeb8181d15aa

I tried to push this on directly on top of this PR but didn't have the permissions to do so.  It appears that `Allow edits from maintainers` is disabled.

**Comment by baby-gnu** [comment]
Note that there is no python `toml` library on `amazonlinux`

```
./bin/kitchen exec default-amazonlinux-2-master-py3 -c 'sudo yum search toml'
-----> Execute command on default-amazonlinux-2-master-py3.
       Failed to set locale, defaulting to C
       Modules compl?mentaires charg?s?: ovl, priorities
       Attention : aucune correspondance trouvée pour : toml
       No matches found
```

**Comment by myii** [comment]
@baby-gnu Thanks for adding the commit.

> Note that there is no python `toml` library on `amazonlinux`

Actually, there is a package available by adding the EPEL repo:

* https://dl.fedoraproject.org/pub/epel/7/x86_64/Packages/p/
  - python36-pytoml-0.1.20-1.el7.noarch.rpm

I'm sure it would be possible to get `amazonlinux` working by adding this repo.  There are examples of doing this inline in other formulas, which I could dig out if required.

**Comment by myii** [comment]
@baby-gnu Re: `amazonlinux-2`, similar to the change made to the [`openvpn-formula`](https://github.com/saltstack-formulas/openvpn-formula/pull/122/files#diff-ea3489002e240c1ef44179e773d6e55d).

**Comment by baby-gnu** [comment]
Thanks @myii, as I'm starting to use the telegraf-formula, I think I can work a little bit more on it. But `amazonlinux` is not my priority for now.

**Pull request merged**